packages/video_player/video_player_web/example/integration_test/pkg_web_tweaks.dart

```diff
@@ -0,0 +1,75 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+@JS()
+library video_player_web_integration_test_pkg_web_tweaks;
+
+import 'dart:js_interop';
+import 'package:web/web.dart' as web;
+
+/// Adds a `controlsList` and `disablePictureInPicture` getters.
+extension NonStandardGettersOnVideoElement on web.HTMLVideoElement {
+  external web.DOMTokenList? get controlsList;
+  external JSBoolean get disablePictureInPicture;
+}
+
+/// Adds a `disableRemotePlayback` getter.
+extension NonStandardGettersOnMediaElement on web.HTMLMediaElement {
+  external JSBoolean get disableRemotePlayback;
+}
+
+/// Defines JS interop to access static methods from `Object`.
+@JS('Object')
+extension type DomObject._(JSAny _) {
+  @JS('defineProperty')
+  external static void _defineProperty(
+      JSAny? object, JSString property, Descriptor value);
+
+  /// `Object.defineProperty`.
+  ///
+  /// See: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/defineProperty
+  static void defineProperty(
+      JSObject object, String property, Descriptor descriptor) {
+    return _defineProperty(object, property.toJS, descriptor);
+  }
+}
+
+/// The descriptor for the property being defined or modified with `defineProperty`.
+///
+/// See: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/defineProperty#description
+extension type Descriptor._(JSObject _) implements JSObject {
+  /// Builds a "data descriptor".
+  factory Descriptor.data({
+    bool? writable,
+    JSAny? value,
+  }) =>
+      Descriptor._data(
+        writable: writable?.toJS,
+        value: value.jsify(),
+      );
+
+  /// Builds an "accessor descriptor".
+  factory Descriptor.accessor({
+    void Function(JSAny? value)? set,
+    JSAny? Function()? get,
+  }) =>
+      Descriptor._accessor(
+        set: set?.toJS,
+        get: get?.toJS,
+      );
+
+  external factory Descriptor._accessor({
```

Comment:
Does this name "work"? How does it align to the JS type?

@srujzs

Reply:
It's an external constructor with named args, so the name is irrelevant to the lowerings. It'll just create an object with the provided arguments like other object literal constructors.

Event if it wasn't an object literal constructor, the name of the constructor doesn't make a difference. We only care about the name of the type for constructors.

Reply:
Cool! TIL

Reply:
At some point I tried to call it `getter` with a `JS('get')` annotation, but I'm not sure that worked as expected (there were some extra changes until I landed on this, so I'm not sure the name of the parameter was an actual isue though)

Reply:
(Confirmed that this:

```dart
  external factory Descriptor._accessor({
    // JSBoolean configurable,
    // JSBoolean enumerable,
    @JS('set')
    JSFunction? setter,
    @JS('get')
    JSFunction? getter,
  });
```

doesn't work!!)

Reply:
Yeah we don't support renaming fields in object literal constructors, but that might be a good idea. The name matters, and if a field needs to be something like `continue`, you're out of luck. Luckily `set` and `get` can be shadowed here.

Reply:
@srujzs should I file an issue for this?

Reply:
sure, go for it! it'll require some moving around of components, but it's good to at least have an issue.

Reply:
Kablamo: https://github.com/dart-lang/web/issues/202